cmd/channel.go

```diff
@@ -45,81 +48,141 @@ func channelCmdChain() *cobra.Command {
-	}
+			tx, err := mazzaroth.Transaction(sender, channelId).Contract(mazzaroth.GenerateNonce(), defaultBlockExpirationNumber).Delete().Sign(pk)
+			if err != nil {
+				return err
+			}
 
-	channelGenCmd := &cobra.Command{
-		Use:   "gen",
-		Short: "generate a mazzaroth channel",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			// Generate Key
-			_, pub, err := crypto.GenerateEd25519KeyPair()
+			client, err := mazzaroth.NewMazzarothClient(mazzaroth.WithAddress(viper.GetString(channelAddress)))
 			if err != nil {
 				return err
 			}
-			fmt.Println("channel address:", crypto.ToHex(pub))
-			// TODO
-			// self signed cert for channel
-			// mazzaroth.io cert generate for channel
+
+			id, _, err := client.TransactionSubmit(cmd.Context(), tx)
+			if err != nil {
+				return err
+			}
+
+			fmt.Println("transaction id:", id)
 			return nil
 		},
 	}
 
 	channelDeployCmd := &cobra.Command{
 		Use:   "deploy",
-		Short: "deploy a channel contract to mazzaroth nodes",
+		Short: "deploy a channel contract to mazzaroth nodes from a given manifest",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			manifestPath := viper.GetString(deploymentManifestPath)
+			manifestPath := viper.GetString(deploymentManifest)
 			// check if file is in default path
 			if _, err := os.Stat(manifestPath); errors.Is(err, os.ErrNotExist) {
 				return errors.New("unable to locate deployment manifest")
 			}
 
+			pk, err := crypto.FromHex(viper.GetString(privateKey))
+			if err != nil {
+				return err
+			}
+
 			manifests, err := manifest.FromFile(manifestPath, "deployment")
 			if err != nil {
 				return err
 			}
 
-			client, err := mazzaroth.NewMazzarothClient(mazzaroth.WithAddress(viper.GetString(address)))
+			client, err := mazzaroth.NewMazzarothClient(mazzaroth.WithAddress(viper.GetString(channelAddress)))
 			if err != nil {
 				return err
 			}
 
-			if err := manifest.ExecuteDeployments(cmd.Context(), manifests, client, "", nil); err != nil {
+			if err := manifest.ExecuteDeployments(cmd.Context(), manifests, client, viper.GetString(publicKey), pk); err != nil {
 				return err
 			}
 			return nil
 		},
 	}
-	channelDeployCmd.Flags().String(deploymentManifestPath, defaultDeploymentManifestPath, "location of mazzaroth channel deployment manifest")
+	channelDeployCmd.Flags().String(deploymentManifest, defaultDeploymentManifestPath, "location of mazzaroth channel deployment manifest")
+
+	channelPauseCmd := &cobra.Command{
+		Use:   "pause",
+		Short: "pause/unpause a channel contract",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			sender, err := xdr.IDFromHexString(viper.GetString(publicKey))
+			if err != nil {
+				return err
+			}
+
+			channelId, err := xdr.IDFromHexString(viper.GetString(channelId))
+			if err != nil {
+				return err
+			}
+
+			tx, err := mazzaroth.Transaction(sender, channelId).Contract(mazzaroth.GenerateNonce(), defaultBlockExpirationNumber).
+				Pause(viper.GetBool(pause)).
+				Sign(nil)
```

Comment:
The private Key needs to be provided here.

Reply:
pushed